src/lib.rs

```diff
@@ -1379,6 +1384,312 @@ mod universe_tests {
         // 1 bit surrounding 'F', and inclusive, are cleared
         assert_eq!(output, 0xFFFFFFE07FFFFFFF);
     }
+
+    #[test]
+    fn verify_fog_circle_bitmap_generation() {
+        let mut uni = generate_test_universe_with_default_params();
+
+        let fog_radius_of_nine = vec![
+            vec![0xf007ffffffffffff],
+            vec![0xe003ffffffffffff],
+            vec![0xc001ffffffffffff],
+            vec![0x8000ffffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x00007fffffffffff],
+            vec![0x8000ffffffffffff],
+            vec![0xc001ffffffffffff],
+            vec![0xe003ffffffffffff],
+            vec![0xf007ffffffffffff]];
+        uni.generate_fog_circle_bitmap(9);
+        assert_eq!(fog_radius_of_nine, uni.fog_circle);
+
+        let fog_radius_of_four = vec![
+            vec![0x83ffffffffffffff],
+            vec![0x01ffffffffffffff],
+            vec![0x01ffffffffffffff],
+            vec![0x01ffffffffffffff],
+            vec![0x01ffffffffffffff],
+            vec![0x01ffffffffffffff],
+            vec![0x83ffffffffffffff],
+        ];
+        uni.generate_fog_circle_bitmap(4);
+        assert_eq!(fog_radius_of_four, uni.fog_circle);
+
+        let fog_radius_of_hunned = vec![
```

Comment:
Impressive, but damn, that's taking up a lot of space in the file... How about a radius 35 circle instead? That is much smaller but still tests multi-u64 rows.

Reply:
Don't want to make it 💯 ? Sure I can trim this down. :)